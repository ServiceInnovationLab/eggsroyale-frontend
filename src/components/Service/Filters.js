import React from 'react';
import '../../styles/Form.css';
import { Route } from 'react-router-dom';

class Filters extends React.Component {

  constructor() {
    super();

    this.categoryChange = this.categoryChange.bind(this);
  }


  componentDidMount(){
    if(this.props.searchVars.category){
      this.categoryChange(this.props.searchVars.category);
    }
  }


  handleChange(event) {
    this.categoryChange(this.props.searchVars.category === event.target.value ? '' : event.target.value);
  }

  categoryChange(category){
    const clone = {...this.props.searchVars};
    clone.category = category;
    this.props.loadResults(clone);
  }

  render(){
    return (
      <div className={'filters'+(this.props.filters.length === 0 ? ' loading' : '') }>
        <nav className="nav">
          {this.props.filters.map((data,index) => {
            return (<Route key={index} render={({ history}) => (
              <button className={this.props.searchVars.category === data.name ? 'selected'  : ''} key={data.num}
                onClick={()=> {
                  history.push((this.props.searchVars.category === data.name ? '' : '/category/'+encodeURIComponent(data.name)));
                  this.categoryChange((this.props.searchVars.category === data.name ? '' : data.name));
                }}>{data.name}</button>)} />
            );
          })}
        </nav>
        <Route render={({history}) => (
          <select value={this.props.searchVars.category} onChange={(event)=> {
            history.push(((this.props.searchVars.category === event.target.value || event.target.value === '') ? '' : '/category/'+encodeURIComponent(event.target.value)));
            this.categoryChange(this.props.searchVars.category === event.target.value ? '' : event.target.value);
          }}>
            <option name="filters" value="" defaultValue>-- Select Category --</option>
            {this.props.filters.map(data => {
              return (<option key={data.num} name="filters">{data.name}</option>);
            })}</select>)} />
      </div>
    );
  }
}

export default Filters;
