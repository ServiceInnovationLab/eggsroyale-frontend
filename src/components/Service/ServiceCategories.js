import React from 'react';
import { Route } from 'react-router-dom';
import {withRouter} from 'react-router';

class ServiceCategories extends React.Component {

  render() {
    return (
      <div className="service-categories">
        <small><b>Service categories: </b></small>
        <small>
          <ul className="list-stripped">
            {this.props.categories.map((data,index) => <Route key={index} render={() => (
              <li className={data === decodeURIComponent(this.props.category) ? 'selected'  : ''}>
                <a href={`/service/${this.props.serviceId}`} onClick={(e)=> {
                  e.preventDefault();
                  this.props.displayServiceDetails(data);
                }} >{data}</a>
              </li>)} />
            )}
          </ul>
        </small>
      </div>
    );
  }
}

export default withRouter(ServiceCategories);
