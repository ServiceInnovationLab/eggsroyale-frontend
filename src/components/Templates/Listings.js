import React from 'react';
import Image from '../Image';
import { connect } from 'react-redux';
import * as actionCreators from '../../actions/index';
import fontawesome from '@fortawesome/fontawesome';
import FontAwesomeIcon from '@fortawesome/react-fontawesome';
import brands from '@fortawesome/fontawesome-free-brands';
import home from '@fortawesome/fontawesome-free-solid/faHome';
import health from '@fortawesome/fontawesome-free-solid/faPlusSquare';
import activities from '@fortawesome/fontawesome-free-solid/faFutbol';
import food from '@fortawesome/fontawesome-free-solid/faCoffee';
import wellbeing from '@fortawesome/fontawesome-free-solid/faLeaf';
import { Link } from 'react-router-dom';
import {loadData, mergeData} from '../../actions/index';

import * as services from '../../csv.json';

fontawesome.library.add(brands, home, health, activities, food, wellbeing);
class Listings extends React.Component {

  constructor(props) {
    super(props);

    this.state = {
      page: window.location.href.split('/').slice(-1)[0],
      results: []
    };
  }

  componentDidMount() {
    this.setState({results: loadData(this.state.page)});
  }

  renderTheme(state) {
    switch(state) {
    case state:
      return state;
    default:
      return 'home';
    }
  }

  renderIcon(state) {
    switch(state) {
    case 'home':
      return 'home';
    case 'food':
      return 'coffee';
    case 'activities':
      return 'futbol';
    case 'wellbeing':
      return 'leaf';
    case 'health':
      return 'plus-square';
    default:
      return 'home';
    }
  }

  text_truncate(str, length, ending) {
    if (length == null) {
      length = 100;
    }
    if (ending == null) {
      ending = '...';
    }
    if (str.length > length) {
      return str.substring(0, length - ending.length) + ending;
    } else {
      return str;
    }
  }

  onlyUnique(value, index, self) {
    const unique = self.indexOf(value) === index;
    return unique;
  }

  render(){
    document.querySelector('body').setAttribute('class',`${this.renderTheme(this.state.page)}-bg`);
    return (
      <div className={`${this.renderTheme(this.state.page)}-bg listing`}>
        <ListingHeader
          page={this.state.page}
          theme={this.renderTheme(this.state.page)}
          icon={this.renderIcon(this.state.page)}
        />
        <div className="container">
          <ListItems
            data={mergeData(services, this.state.results)}
            theme={this.renderTheme(this.state.page)}
            page={this.state.page}
          />
        </div>
      </div>
    );
  }
}

const ListItems = props => {
  return <ul className="list-stripped">
    {services.length > 0 && services.filter(x => x.CATEGORY === props.page).map((item, key) => {
      return <li key={key} className={props.theme}>
        <a href={`#/${props.page}/0000${key+1}`} className="service">
          <Image src="http://placekitten.com/200/300" alt="kitten" />
          <span className="listing-details">
            <h3>{item.SERVICE_NAME}</h3>
            <p>{item.PROVIDER_NAME}</p>
          </span>
        </a>
      </li>;
    })}
  </ul>;
};

const ListingHeader = props => {
  return <header className={props.theme}>
    <Link to="/" className="back-link">
      <span className="arrow arrow-left"></span>
      <span className="aria-hidden">Navigate to home</span>
    </Link>master
    <FontAwesomeIcon icon={props.icon} />
    <h2>{props.page}</h2>
  </header>;
};

const mapStateToProps=(state)=>{
  return state;
};

export default connect (mapStateToProps, actionCreators)(Listings);
