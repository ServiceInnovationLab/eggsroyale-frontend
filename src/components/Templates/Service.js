import React from 'react';
import Image from '../Image';
import fontawesome from '@fortawesome/fontawesome';
import FontAwesomeIcon from '@fortawesome/react-fontawesome';
import brands from '@fortawesome/fontawesome-free-brands';
import home from '@fortawesome/fontawesome-free-solid/faHome';
import health from '@fortawesome/fontawesome-free-solid/faPlusSquare';
import activities from '@fortawesome/fontawesome-free-solid/faFutbol';
import food from '@fortawesome/fontawesome-free-solid/faCoffee';
import bookmark from '@fortawesome/fontawesome-free-solid/faBookmark';
import axios from 'axios';
fontawesome.library.add(brands, home, health, activities, food, bookmark);

class Service extends React.Component {
  constructor(props) {
    super(props);

    const loc = window.location.href.split('/');
    this.state = {
      page: loc.slice(-2)[0],
      service: loc.slice(-1)[0],
      results: []
    };
  }

  loadService(){
    // const GLOBAL_FILTER = 'community services card';
    const RESOURCE_ID = process.env.REACT_APP_API_RESOURCE_ID;
    const API_PATH = process.env.REACT_APP_API_PATH;
    // const CATEGORY = this.getCategory(this.state.page);
    let fields = '*';

    let where = `WHERE "FSD_ID" = '${this.state.service}'`;

    // alert(CATEGORY.split(',')[0])

    let sql =`SELECT DISTINCT ${fields} FROM "${RESOURCE_ID}" ${where} LIMIT 1`;
    sql =  encodeURI(sql);
    let url = `${API_PATH}datastore_search_sql?sql=${sql}`;
    return axios.get(url).then((response)=>{
      this.setState({results: response.data.result.records[0]});
    });
  }

  renderTheme(state) {
    switch(state) {
    case state:
      return state;
    default:
      return 'home';
    }
  }
  componentDidMount() {
    return this.loadService();
  }
  render() {
    return (
      <div className={`${this.renderTheme(this.state.page)}-bg listing service`}>
        <Header />
        <Subheader
          theme={this.renderTheme(this.state.page)}
          image="http://placekitten.com/200/300"
          service={this.state.results.PROVIDER_NAME}
          serviceDesc={this.state.results.SERVICE_DETAIL}
          page={this.state.page}
        />
        <div className="container-inner">
          <p>{this.state.results.ORGANISATION_PURPOSE}</p>
          <p>{this.state.results.PROVIDER_CONTACT_AVAILABILITY}</p>
          <p>{this.state.results.PHYSICAL_ADDRESS}</p>
          <p>{this.state.results.PHYSICAL_DISTRICT}</p>
          <p>{this.state.results.PHYSICAL_REGION}</p>
          <p>{this.state.results.POSTAL_ADDRESS}</p>
        </div>
      </div>
    );
  }
}

const Header = () => {
  return (
    <header className="home-header compact">
      <a href="/" className="back-link">
        <span className="arrow arrow-left arrow-sm"></span>
        <FontAwesomeIcon icon="home" />
        <span><p>Home</p></span>
        <span className="aria-hidden">Navigate to home</span>
      </a>
    </header>
  );
};

function addBookmark(page) {
  let bookmarks = window.localStorage.getItem("csc-bookmarks");
  return bookmarks+`,${window.location.href}`;
}

const Subheader = props => {
  return (
    <div className={`${props.theme} sub-header`}>
      <div className="container">
        <Image src={props.image} alt={props.alt} />
        <header className={props.theme} style={{display: 'inline-block'}}>
          <FontAwesomeIcon icon="bookmark" onClick={()=>window.localStorage.setItem('csc-bookmarks', addBookmark(props.page))} />
          <h2>{props.service}</h2>
          <p>{props.serviceDesc}</p>
        </header>
      </div>
    </div>
  );
};

export default Service;
