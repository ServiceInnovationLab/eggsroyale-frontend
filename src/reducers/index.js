let defaultState = {
  filter: [],
  results: [],
  result: [],
  searchVars: {
    category: '',
    keyword: '',
    address: '',
    addressLatLng: {},
    radius: 50000
  },
  serviceDetails: [],
  noSearchVars: false,
  totalResults: 0,
  itemsLoading: false,
  hasSearched: false
};

const mainReducer = (state = defaultState, action) => {
  switch (action.type) {
  case 'LOAD_RESULTS':
    return {
      ...state,
      itemsLoading: action.loading
    };
  case 'SHOW_FILTERS':
    return {
      ...state,
      filter: action.filters
    };
  case 'SHOW_RESULTS':
    return {
      ...state,
      results: action.results,
      searchVars: action.searchVars,
      noSearchVars: action.noSearchVars,
      totalResults: action.totalResults,
      hasSearched: true,
      itemsLoading: false
    };
  case 'SHOW_SERVICE':
    return {
      ...state,
      result: action.result,
      itemsLoading: false
    };
  case 'SHOW_SERVICE_DETAILS':
    return {
      ...state,
      serviceDetails: action.serviceDetails
    };
  case 'CHANGE_CATEGORIES':
    return {
      ...state,
      searchVars: action.searchVars
    };
  default:
    return {
      ...state
    };
  }
};

export default mainReducer;
