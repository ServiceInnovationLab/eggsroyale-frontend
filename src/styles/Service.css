.leaflet-container {
  margin-top: 10px; }

a {
  color: #35a744; }

.service {
  border-bottom: 1px solid #c4c4c4;
  margin-bottom: 50px;
  padding-bottom: 30px; }

.service,
.service-info {
  padding-right: 5%; }
  .service a,
  .service-info a {
    text-decoration: none; }
    .service a h3,
    .service-info a h3 {
      color: #35a744; }
    .service a .telephone,
    .service-info a .telephone {
      color: #35a744; }
  .service dt,
  .service-info dt {
    clear: left;
    float: left;
    font-weight: bold;
    width: 160px; }
  .service dt::after,
  .service-info dt::after {
    content: ":"; }
  .service dd,
  .service-info dd {
    margin-bottom: 7px;
    margin-left: 160px;
    margin-right: 0;
    margin-top: 0;
    padding-bottom: 0;
    padding-left: 0.5em;
    padding-right: 0;
    padding-top: 0; }

.search-result-hero {
  border-left: 4px solid #35a744;
  padding-bottom: 5px;
  padding-left: 21px;
  padding-top: 3px; }
  .search-result-hero a {
    color: #35a744; }
  .search-result-hero h2 {
    font-size: 24px;
    margin: 7px 0 0; }
  .search-result-hero p {
    margin-bottom: 3px;
    margin-top: 0; }
  .search-result-hero ul {
    margin-bottom: 0;
    margin-top: 1em; }

/*# sourceMappingURL=Service.css.map */
