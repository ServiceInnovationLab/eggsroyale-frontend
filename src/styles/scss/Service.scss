$primary-color: #35a744;
$secondary-color: #c4c4c4;

.leaflet-container {
  margin-top: 10px;
}

a {
  color: $primary-color;
}

.service {
  border-bottom: 1px solid $secondary-color;
  margin-bottom: 50px;
  padding-bottom: 30px;
}

.service,
.service-info {
  padding-right: 5%;

  a {
    text-decoration: none;

    h3 { color: $primary-color; }

    .telephone { color: $primary-color; }
  }

  dt {
    clear: left;
    float: left;
    font-weight: bold;
    width: 160px;
  }

  dt::after {
    content: ":";
  }

  dd {
    margin-bottom: 7px;
    margin-left: 160px;
    margin-right: 0;
    margin-top: 0;
    padding-bottom: 0;
    padding-left: 0.5em;
    padding-right: 0;
    padding-top: 0;
  }
}

.search-result-hero {
  border-left: 4px solid $primary-color;
  padding-bottom: 5px;
  padding-left: 21px;
  padding-top: 3px;

  a { color: $primary-color; }

  h2 {
    font-size: 24px;
    margin: 7px 0 0;
  }

  p {
    margin-bottom: 3px;
    margin-top: 0;
  }

  ul {
    margin-bottom: 0;
    margin-top: 1em;
  }
}

