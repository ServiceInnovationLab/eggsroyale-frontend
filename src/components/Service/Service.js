import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import '../../styles/Service.css';
import ServiceDetail from './ServiceDetail';
import ServiceContactDetail from './ServiceContactDetail';

class Service extends Component {

  render() {
    return (
      <div>
        <div className="service">
          <div className="search-result-hero">
            <h2><Link to={`/service/${this.props.results.FSD_ID}`}>{this.props.results.PROVIDER_NAME}</Link></h2>
            {this.props.results.DISTANCE && <small>{((this.props.results.DISTANCE*1)/1000).toFixed(1)} km</small>}
            <ServiceContactDetail phone={this.props.results.PUBLISHED_PHONE_1} email={this.props.results.PUBLISHED_CONTACT_EMAIL_1} hours={this.props.results.PROVIDER_CONTACT_AVAILABILITY} />
          </div>
          <ServiceDetail results={this.props.results} changeCategory={this.props.changeCategory} searchVars={this.props.searchVars} serviceId={this.props.serviceId} loadimmediately={false} loadResults={this.props.loadResults} />
        </div>
      </div>
    );
  }
}

export default Service;
