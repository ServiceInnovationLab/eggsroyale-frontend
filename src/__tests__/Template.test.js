import React from 'react';
import renderer from 'react-test-renderer';
import Service from '../components/Templates/Service';
import Listings from '../components/Templates/Listings';

const Components = [
  Service,
  Listings
];
import { Provider } from 'react-redux';
import { createStore, applyMiddleware } from 'redux';
import reducers from '../reducers/index';
import thunk from 'redux-thunk';
const store = createStore(reducers, applyMiddleware(thunk));

describe('components', function() {
  'use strict';
  Components.map(item=> {
    describe(`<${item} />`, function() {
      it('renders correctly', function() {
        const TagName = item;
        let tree = renderer.create(<Provider store={store}><TagName /></Provider>).toJSON();
        expect(tree).toMatchSnapshot();
      });
    });
  });
});
