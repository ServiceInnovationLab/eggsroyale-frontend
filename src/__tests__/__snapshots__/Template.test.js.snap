// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components <function Connect(props, context) {
        _classCallCheck(this, Connect);

        var _this = _possibleConstructorReturn(this, _Component.call(this, props, context));

        _this.version = version;
        _this.state = {};
        _this.renderCount = 0;
        _this.store = props[storeKey] || context[storeKey];
        _this.propsMode = Boolean(props[storeKey]);
        _this.setWrappedInstance = _this.setWrappedInstance.bind(_this);

        (0, _invariant2.default)(_this.store, 'Could not find "' + storeKey + '" in either the context or props of ' + ('"' + displayName + '". Either wrap the root component in a <Provider>, ') + ('or explicitly pass "' + storeKey + '" as a prop to "' + displayName + '".'));

        _this.initSelector();
        _this.initSubscription();
        return _this;
      } /> renders correctly 1`] = `
<div
  className="about:blank-bg listing"
>
  <header
    className="about:blank"
  >
    <a
      className="back-link"
      href="/"
      onClick={[Function]}
    >
      <span
        className="arrow arrow-left"
      />
      <span
        className="aria-hidden"
      >
        Navigate to home
      </span>
    </a>
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-home fa-w-18 "
      data-icon="home"
      data-prefix="fas"
      role="img"
      style={Object {}}
      viewBox="0 0 576 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M488 312.7V456c0 13.3-10.7 24-24 24H348c-6.6 0-12-5.4-12-12V356c0-6.6-5.4-12-12-12h-72c-6.6 0-12 5.4-12 12v112c0 6.6-5.4 12-12 12H112c-13.3 0-24-10.7-24-24V312.7c0-3.6 1.6-7 4.4-9.3l188-154.8c4.4-3.6 10.8-3.6 15.3 0l188 154.8c2.7 2.3 4.3 5.7 4.3 9.3zm83.6-60.9L488 182.9V44.4c0-6.6-5.4-12-12-12h-56c-6.6 0-12 5.4-12 12V117l-89.5-73.7c-17.7-14.6-43.3-14.6-61 0L4.4 251.8c-5.1 4.2-5.8 11.8-1.6 16.9l25.5 31c4.2 5.1 11.8 5.8 16.9 1.6l235.2-193.7c4.4-3.6 10.8-3.6 15.3 0l235.2 193.7c5.1 4.2 12.7 3.5 16.9-1.6l25.5-31c4.2-5.2 3.4-12.7-1.7-16.9z"
        fill="currentColor"
        style={Object {}}
      />
    </svg>
    <h2>
      about:blank
    </h2>
  </header>
  <div
    className="container"
  >
    <ul
      className="list-stripped"
    />
  </div>
</div>
`;

exports[`components <function Service(props) {
    _classCallCheck(this, Service);

    var _this = _possibleConstructorReturn(this, (Service.__proto__ || Object.getPrototypeOf(Service)).call(this, props));

    var loc = window.location.href.split('/');
    _this.state = {
      page: loc.slice(-2)[0],
      id: loc.slice(-1)[0],
      service: loc.slice(-1)[0],
      results: []
    };
    return _this;
  } /> renders correctly 1`] = `
<div
  className="about:blank-bg listing service"
>
  <header
    className="home-header compact"
  >
    <a
      className="back-link"
      href="/"
      onClick={[Function]}
    >
      <span
        className="arrow arrow-left arrow-sm"
      />
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-home fa-w-18 "
        data-icon="home"
        data-prefix="fas"
        role="img"
        style={Object {}}
        viewBox="0 0 576 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M488 312.7V456c0 13.3-10.7 24-24 24H348c-6.6 0-12-5.4-12-12V356c0-6.6-5.4-12-12-12h-72c-6.6 0-12 5.4-12 12v112c0 6.6-5.4 12-12 12H112c-13.3 0-24-10.7-24-24V312.7c0-3.6 1.6-7 4.4-9.3l188-154.8c4.4-3.6 10.8-3.6 15.3 0l188 154.8c2.7 2.3 4.3 5.7 4.3 9.3zm83.6-60.9L488 182.9V44.4c0-6.6-5.4-12-12-12h-56c-6.6 0-12 5.4-12 12V117l-89.5-73.7c-17.7-14.6-43.3-14.6-61 0L4.4 251.8c-5.1 4.2-5.8 11.8-1.6 16.9l25.5 31c4.2 5.1 11.8 5.8 16.9 1.6l235.2-193.7c4.4-3.6 10.8-3.6 15.3 0l235.2 193.7c5.1 4.2 12.7 3.5 16.9-1.6l25.5-31c4.2-5.2 3.4-12.7-1.7-16.9z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
      <span>
        <p>
          Home
        </p>
      </span>
      <span
        className="aria-hidden"
      >
        Navigate to home
      </span>
    </a>
  </header>
</div>
`;
